ui/src/app/nav.tsx

```diff
@@ -24,12 +33,20 @@ export function Aside({ isOpen, setIsOpen }: { isOpen: boolean, setIsOpen: Dispa
                         setCSRPEMString(e.target.result.toString());
                     }
                 }
-        };
-        reader.readAsText(file);
+            };
+            reader.readAsText(file);
+        }
+    };
+    const CSRValidationElement = function () {
+        try {
+            extractCSR(CSRPEMString)
+        } catch (e) {
+            return <div><i className="p-icon--error"></i>Invalid CSR</div>
```

Comment:
I dislike that we can submit an invalid CSR or an empty one. We could disable/grey out the submit button until a valid CSR was provided.

Reply:
done